prepare_data/src/test/groovy/org/orbisgis/bdtopo/BDTopoGISLayersTest.groovy

```diff
@@ -4,27 +4,42 @@ import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.condition.EnabledIfSystemProperty
 import org.orbisgis.PrepareData
+import org.orbisgis.common.InputDataFormattingTest
 import org.orbisgis.datamanager.h2gis.H2GIS
 
 import static org.junit.jupiter.api.Assertions.assertNotNull
-import static org.junit.jupiter.api.Assertions.assertNull
 import static org.junit.jupiter.api.Assertions.assertTrue
 
 class BDTopoGISLayersTest {
 
-    private final static String bdTopoDb = System.getProperty("user.home") + "/myh2gisbdtopodb.mv.db"
+    private final static File bdTopoDb = new File("./target/myh2gisbdtopodb.mv.db")
 
     @BeforeAll
     static void init(){
-        System.setProperty("test.bdtopo", Boolean.toString(new File(bdTopoDb).exists()))
+        boolean isFile = InputDataFormattingTest.getResource("myh2gisbdtopodb.mv.db") != null
+        System.setProperty("test.bdtopo", Boolean.toString(isFile))
+        if(isFile) {
+            InputStream is = null
+            OutputStream os = null
+            try {
+                is = InputDataFormattingTest.getResourceAsStream("myh2gisbdtopodb.mv.db")
+                os = new FileOutputStream(bdTopoDb)
+                byte[] buffer = new byte[1024]
+                int length
+                while ((length = is.read(buffer)) > 0) {
+                    os.write(buffer, 0, length)
+                }
+            } finally {
+                is.close()
+                os.close()
+            }
```

Comment:
I do not really understand what you are doing from line 21 up to the end. Could you explain or comment in the code ?

Reply:
I have changed this piece of code in order to use groovy code and I've added comments.